service/task.go

```diff
@@ -173,7 +173,7 @@ func (uis *UIServer) taskPage(w http.ResponseWriter, r *http.Request) {
 		if taskFromDb != nil {
 			projCtx.Task = taskFromDb
 			archived = true
-		} else if execution != projCtx.Task.Execution {
+		} else if execution != projCtx.Task.Execution && !projCtx.Task.IsPartOfDisplay() {
```

Comment:
if the page is showing an archived execution of an execution task with the task id from the old tasks collection won't this cause all the links to point to the same execution (the currently selected execution)?

Reply:
I don't quite follow. Archived executions each have unique IDs, and those should be looked up from the old collection and loaded into projCtx.Task

Reply:
Following this path:
1. I go to a previous execution of a display task ([this page ](https://evergreen.mongodb.com/task/mongodb_mongo_master_enterprise_rhel_62_64_bit_display_rollback_fuzzer_unclean_shutdowns_patch_a368e90685fdaa5d855ecd54769f5ca0c09d4378_5d04308d850e6160f69b880e_19_06_14_23_46_06/5##%257B%2522compare%2522%253A%255B%257B%2522hash%2522%253A%2522a368e90685fdaa5d855ecd54769f5ca0c09d4378%2522%257D%255D%257D) for instance), let's say the 6th execution. The task id is from the old tasks collection (with `_6` appended).
2. I click a link to an execution task ([this one](https://evergreen.mongodb.com/task/mongodb_mongo_master_enterprise_rhel_62_64_bit_rollback_fuzzer_unclean_shutdowns_00_enterprise_rhel_62_64_bit_patch_a368e90685fdaa5d855ecd54769f5ca0c09d4378_5d04308d850e6160f69b880e_19_06_14_23_46_06_5/5##%257B%2522compare%2522%253A%255B%257B%2522hash%2522%253A%2522a368e90685fdaa5d855ecd54769f5ca0c09d4378%2522%257D%255D%257D) for instance). This takes me to the task page for the execution task, with the task being pulled from the old task collection (with `_6` appended), and the execution number parameter set to match the execution of the task.

If I navigate to a previous execution of the task, let's say the 1st, it will be asking for the 1st execution of the old_task, which previously returned an error that it didn't exist. If this is allowed through it will return the task from the context, which will be the 6th execution. However, what I really wanted was the task from the old tasks collection with `_1` appended.

Reply:
Gotcha, my approach is incorrect. I changed this to instead to not have the execution in the task ID